src/AuthService/Application/External/Infrastructure/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Linq.Expressions;
+using Domain.Abstractions;
+using Domain.Entities.Base;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace Infrastructure.Repositories;
+
+public abstract class BaseRepository<T>(AuthDbContext dbContext) : IBaseRepository<T> where T : class, IEntity
+{
+    public async Task<List<T>> FindAllAsync(bool trackChanges, CancellationToken cancellationToken = default)
+    {
+        return !trackChanges
+            ? await dbContext.Set<T>().AsNoTracking().ToListAsync(cancellationToken)
+            : await dbContext.Set<T>().ToListAsync(cancellationToken);
+    }
+
+    public async Task<List<T>> FindByConditionAsync(Expression<Func<T, bool>> expression, bool trackChanges, CancellationToken cancellationToken = default)
```

Comment:
I would suggest either to keep the trackChanges logic not here or to provide some default value